packages/evm/src/opcodes/functions.ts

```diff
@@ -632,6 +632,20 @@ export const handlers: Map<number, OpHandler> = new Map([
       runState.stack.push(runState.interpreter.getBlockBaseFee())
     },
   ],
+  // 0x49: DATAHASH
+  [
+    0x49,
+    function (runState) {
+      const index = runState.stack.pop()
+      if (
+        runState.env.versionedHashes?.length !== undefined &&
+        runState.env.versionedHashes.length > Number(index)
```

Comment:
Do we want to initialize `env.versionedHashes` with an empty array and then we can skip this double check here?

Reply:
Good idea.  Done.